r_scripts/data_wrangle/asthma_cohort.R

```diff
@@ -0,0 +1,122 @@
+# ------------------------------------------------------------------------------
+# Title: Basic data wrangling  for 2013 Oregon data and seperating by ICD9 lists
+# Author: Jingyang Liu
+# Date Created: Mar 03, 2016
+# Date Checked and Modified: Nov 20, 2017
+# R Version: 3.3.2 (now 3.4.2)
+# ------------------------------------------------------------------------------
+
+# This file is for filtering saba outcomes.
+# Also it separates the data into 12 disease files due to the ICD9 lists.
+# And combine a asthma at-risk data set with ICD9 of asthma or saba.
+# original data sets and instrunction files such as index are stored in data_original file,
+# the new data sets we made are stored in data_new, which contain health and smoke
+# data sets. Now the outcomes are all stored in data_new/health.
+
+# This runs on server because of its huge computing volume.
+
+
+# library
+library(dplyr)
+library(data.table)
+library(readxl) 
+library(readr)
+
+### Prepare work
+
+# import original data ---------------------------------------------------------
+read_path <- paste0("../../../data/data_original/gan_episodes_of_care.txt") # 77million
+start_time <- Sys.time()
+oregon_df <- fread(read_path, sep = "|", showProgress = T) 
+stop_time <-  Sys.time() - start_time 
+# time it took
+stop_time # 7 mins
+
+
+# basic cleaning ---------------------------------------------------------------
```

Comment:
This is a step that doesn't need to happen. You are reading in the txt data file, saving as a csv, then reading in again. It's not efficient. remove this 'basic cleaning' step.